platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -681,27 +719,46 @@ export default class HangingProtocolService extends PubSubService {
    * simpler than recomputing the entire protocol.
    */
   public getComputedOptions(
-    options: Record<string, unknown>,
+    options: Record<string, unknown> | Array<Record<string, unknown>>,
     displaySetUIDs: string[]
   ) {
+    if (Array.isArray(options)) {
+      return options.map(option =>
+        this.getComputedOptions(option, displaySetUIDs)
+      );
+    }
+
     const computed = { ...options };
-    let displaySets;
-    for (const key in computed) {
-      const value = computed[key];
-      if (!value) continue;
+
+    const processValue = (value: any, parent: any, key: string) => {
```

Comment:
You should check here for whether options has a custom attribute right here - and then call getComputedOptions recursively rather than processValue.  That allows you to replace the entire options, eg:
viewportOptions: { custom: 'myViewportOptions' }